pinery-lims/src/main/java/ca/on/oicr/pinery/lims/LimsAttribute.java

```diff
@@ -75,23 +76,23 @@ public String asString(boolean useEnumToString) {
     }
 
     private static void validate() {
-        Set<String> ins = new HashSet<>();
-        Set<String> outs = new HashSet<>();
-        for (SampleAttribute sa : SampleAttribute.values()) {
-            if (Sets.intersection(ins, sa.in).isEmpty()) {
-                ins.addAll(sa.in);
+        Set<String> allTerms = new HashSet<>();
+        Set<String> allOutputTerms = new HashSet<>();
+        for (LimsAttribute la : LimsAttribute.values()) {
+            if (Sets.intersection(allTerms, la.inputTerms).isEmpty()) {
+                allTerms.addAll(la.inputTerms);
             } else {
-                throw new RuntimeException("Duplicate: [" + sa.in.toString() + "]");
+                throw new IllegalArgumentException("Duplicate: [" + la.inputTerms.toString() + "]");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Duplicate: [" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)